build_docs

```diff
@@ -0,0 +1,177 @@
+import subprocess
+from sys import platform, stdout
+import time
+
+DOCKER_BUILD_QUIET_TIME = 3 # seconds
+
+DIR = dirname(realpath(__file__))
+logger = logging.getLogger('build_docs')
+
+def build_docker_image():
+    logger.info('Building the docker image that will build the docs. Expect '
+        + 'this to take somewhere between a hundred milliseconds and '
+        + 'five minutes.')
+    with open(join(DIR, 'Dockerfile')) as dockerfile:
+        # We attempt to spool up the output from docker build so we can hide it
+        # if the command is successful *and* runs quickly. If it takes longer
+        # than the DOCKER_BUILD_QUIET_TIME then we log all of the output. I
+        # know this is a little sneaky, but folks run this from the command
+        # line so frequently that they'll appreciate cleaner output. I hope.
+        cmd = ["docker", "image", "build", "-t", "elastic/docs_build", "-"]
+        build = subprocess.Popen(
+            cmd,
+            stdin = dockerfile,
+            stdout = subprocess.PIPE,
+            stderr = subprocess.STDOUT)
+        docker_logger = logging.getLogger('docker build')
+        start_logging_at = time.time() + DOCKER_BUILD_QUIET_TIME
+        acc = []
+        while True:
+            line = build.stdout.readline()
+            if line != '':
+                line = line.rstrip()
+                if time.time() >= start_logging_at:
+                    for line in acc:
+                        docker_logger.info(line)
+                    acc = []
+                    docker_logger.info(line)
+                else:
+                    acc.append(line)
+            else:
+                break
+        build.wait()
+        if build.returncode != 0:
+            for line in acc:
+                docker_logger.error(line)
+            raise subprocess.CalledProcessError(build.returncode, cmd)
+
+def run_build_docs(args):
+    docker_args = []
+    build_docs_args = []
+
+    # Remove the container immediately when we're done building the docs
+    docker_args.append('--rm')
+    # Make sure we create files as the current user because that is what
+    # folks that use build_docs.pl expect.
+    docker_args.extend(['--user', '%d:%d' % (getuid(), getgid())])
+    # Running read-only with a proper tmp directory gives us a little
+    # performance boost and it is simple enough to do.
+    docker_args.extend(['--read-only', '--tmpfs', '/tmp'])
+    # Mount the docs build code so we can run it!
+    docker_args.extend(['-v', '%s:/docs_build:cached' % DIR])
+    # Seccomp adds a *devestating* performance overhead if you happen
+    # to have it installed.
+    docker_args.extend(['--security-opt', 'seccomp=unconfined'])
+
+    if stdout.isatty:
+        # Emulate a terminal so things like ctrl-c work.
+        docker_args.append('-it')
+
+    saw_out = False
+    resource_count = 0
+    a = 0
+    while a < len(args):
+        build_docs_args.append(args[a])
```

Comment:
I guess this technique will break if people use GNU style longopts with `=` in them, but maybe `build_docs.pl` doesn't support them anyway? (And maybe I'm the only freak who uses them).

Reply:
Yeah, it doesn't work. And the perl script does. Let me see what I can do about that.

Reply:
It won't pick up the single character alias style options either but the perl script doesn't support any of those.